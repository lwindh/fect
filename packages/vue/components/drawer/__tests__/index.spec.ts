import { mount } from '@vue/test-utils'
import Drawer from '../index'

describe('Drawer', () => {
  it('should be render as  a elemnt', () => {
    const wrapepr = mount(Drawer)
    expect(() => wrapepr.unmount()).not.toThrow()
  })
  it('should be support normal props', async () => {
    const wrapper = mount(
      {
        data() {
          return {
            show: false,
            visible: false,
            disableOverlayClick: false,
            overlay: true,
            placement: 'right',
          }
        },
        components: {
          [Drawer.name]: Drawer,
        },
        template: `
      <div class="container">
       <fe-drawer 
        :placement="placement"
        v-model="visible"
        v-if="show"
        teleport=".container"
        :overlay="overlay"
        :disable-overlay-click="disableOverlayClick"
       >
        <span>
          Test
        </span>
       </fe-drawer>
      </div>
      `,
      },
      { attachTo: document.body }
    )

    await wrapper.setData({ visible: true })
    await wrapper.setData({ show: true })
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setData({ overlay: false })
    expect(wrapper.find('.fect-teleport__overlay').exists()).toBe(false)
    const el = wrapper.find('.fect-drawer__root')
    await el.trigger('click')
    expect(wrapper.vm.visible).toBe(false)
    await wrapper.setData({ visible: true })
    await wrapper.setData({ disableOverlayClick: true })
    await el.trigger('click')
    expect(wrapper.vm.visible).toBe(true)
    const dom = wrapper.find('.fect-drawer__wrapper')
    await dom.trigger('click')
    expect(wrapper.vm.visible).toBe(true)
  })
})
