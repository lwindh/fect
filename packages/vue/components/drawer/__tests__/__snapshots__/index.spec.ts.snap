// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should be support normal props 1`] = `
<div class="container">
  <!--teleport start-->
  <!--teleport end-->
  <transition-stub>
    <div class="fect-teleport__overlay"></div>
  </transition-stub>
  <transition-stub>
    <div class="fect-drawer__root" role="popup" style="--drawer-transfrom: translate3d(100%, 0, 0);">
      <div class="fect-drawer__wrapper fect-drawer__wrapper--right round" role="dialog"><span> Test </span></div>
    </div>
  </transition-stub>
</div>
`;
